### ISSUE TITLE
The possibility to change speed and power breaks the game mechanics

### ISSUE THREAD

**Issue by jensMF** [dulvui/pocket-broomball#39]
The new functionality to change speed and strength renders the wohle game mechanics useless.
I can now just make my team the strongest in the whole tournament and all others the weakest to get enough money to buy everything in the game whitout any effort. The different prices for the teams are not understandable this way, too. Additionally I can unlock all arcade points, too.

May be it would be a good idea to have "arcade money" and "tournament money" anyways.

I think an option could be that it is not possible to change the values of not played teams. Then one could need to upgrade speed and strength like in arcade mode and all teams cost the same. Additionally one could say the bought values are maximas. Or one could buy points to add to the team one wants to play. Or one could only modify the points of the team one wants to play to the maximum of the strongest team one have bought in the tournament.

Or one could buy points to add them to the teams but the actual default values are the minimum.

Or one could buy points for the tournaments and can modify the amount of points one have bought. To explain a bit more what I mean: i. e. when I have 10 points I can add or subtract 10 Points in the whole tournament. I. e. I could add 3 points to the strength of India 3 to its speed and subtract 2 points of speed and 2 points of strength from Canada.

Just some Ideas to keep a challenge in the game.

**Comment by dulvui** [comment]
Thank you a lot for this valuable feedback. I was afraid that this feature ruins the game economy.

The main idea behind this is to let players change the values of the teams, to adapt real world abilities or simply to make one favorites team stronger. The default values of the game are picked by me and purely "random", since I haven't yet found a good source for Broomball tournaments data, to be able to adapt the teams abilities every year, depending on the performance in the previous season. Even if I would find something that can help me with this, there are still some issues with teams that might not even exist anymore. So I would have to remove them, but I would prefer to not do that.

So this was an "easy fix" but surely not the best way. Firstly I wanted to allow changes only for teams one has already unlocked, but then I went for the easy way of simply allowing to change all teams.
I also thought about something like your last suggestion, that changing values costs money, but then when you reset the teams to default (doable in the settings), the money spent is lost.

Easily understandable and also quite easy to implement could be that you first need to unlock all teams of a tournament, before you can modify the values. So the game economy from before is still the same. The only issue I would see there is that if a new team gets added to a league. Would you keep the ability to change teams or would it reset and unlock again, once the new team has be unlocked too?
What would you think about this suggestion and how would you solve the "new teams" problem?

So I would prefer one of this two options:
1) Only change already unlocked teams, like you suggested. But someone could simply buy the cheapest team and make it strong. But at least he needs first to unlock all other teams, to make them weak.
2) Allow changing of teams only once all teams of a tournament are unlocked. So the "old" game style is restored and once you "unlocked all teams" a league, you're free to change however you want. There just remains the "new teams" problem as mentioned before, but that anyway does't happen that often.

I never thought of having separate arcade and tournament money, this is really a good idea. But I feel that it is too late to implement that now, maybe in Pocket Broomball 2, if that ever happens ;-)

**Comment by jensMF** [comment]
I would say either

1. and the best team one bought in the league is the maximum one can raise the values to. Additionally one could add the feature that one can modify the values freely once one unlocked all teams of the league (because some leagues do not have a team with maximum values) or
2. and when a new team is added, all teams in that league are reset to default and one would need to buy the new team.

**Comment by dulvui** [comment]
I like that idea, I just need to find a good way to explain it to the users.
Maybe a popup window that shows this info, when someone clicks on the plus button?

**Comment by jensMF** [comment]
Good idea. But I would try to make the ui self explanatory. Only show the plus and minus buttons, when the value is editable. When the (actual) maximum for a value is reached, hide the plus, when the minimum is reached, hide the minus.
Additional idea for the 1. idea: In case, that the plus button is not hidden, when a user tries to raise a value of a team to a higher value than the maximum unlocked (but not if this is the maximum for the value in the game), explain it.

In addition to what I wrote for idea 1 one could add that not only the best unlocked team defines the maximum value but also the worst unlocked team defines the minimum (until all teams of the league are unlocked)

**Comment by jensMF** [comment]
And my be add a help section where the details are explained, in case someone is interested in the rule behind it.

**Comment by dulvui** [comment]
This repo is getting archived here on Github and moved to Codeberg https://codeberg.org/dulvui/pocket-broomball/
Please open any new issues there, thank you!
